spec-0010/index.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "SPEC 10 — Changelog"
+number: 10
+date: 2024-06-03
+author:
+  - "Inessa Pawson <inessapawson@gmail.com>"
+discussion: https://discuss.scientific-python.org/t/
+endorsed-by:
+---
+
+## Description
+
+SPEC 10 provides guidelines and best practices for maintaining a changelog file
+for libraries in the Scientific Python ecosystem.
+
+### Core Project Endorsement
+
+<!--
+Briefly discuss what it means for a core project to endorse this SPEC.
+-->
+
+### Ecosystem Adoption
+
+The endorsement of this SPEC signifies your project's support for the guidelines 
+laid out in the document.
+
+## Implementation
+
+Keeping a clear, organized, and human-readable record of notable changes for each 
+version of a Python library is essential for the project’s maintenance and sustainability. 
+It promotes transparency and trust between the project’s leadership team, developers, and users.
+
+### What Is a Changelog?
+
+A changelog is a file which contains a curated, chronologically ordered list of 
+notable changes for each version of a project.
+
+### Guiding Principles
+
+- Changelogs are meant for humans, not machines.
+- Every version should have an entry.
+- Group similar types of changes together.
+- Make versions and sections linkable.
+- List the most recent version first.
+- Display the release date for each version.
+- Indicate if you follow Semantic Versioning.
+
+### Types of Changes
+
+- **Added**: New features.
+- **Changed**: Modifications in existing functionality.
+- **Deprecated**: Soon-to-be-removed features.
+- **Removed**: Features that are now removed.
+- **Fixed**: Bug fixes.
+- **Security**: Vulnerability fixes.
```

Comment:
I find that the distinction between added, changed, and removed is often very blurry for me. I've liked the following categorization instead:

- New features (stuff that strictly extends existing API)
- Changed API (removed, deprecated & changed)
- Fixed (possibly group bugs and security here)
- Documentation
- Infrastructure (tooling around the library)
- Maintenance (internal stuff not affecting users)

This is merely a suggestion from personal experience, curious what you think. In any case, I imagine that this document will probably word this as a suggestion anyway in the final state.

Reply:
Do you separate, "Changed API" into those three (removed, deprecated & changed) subtitles? We have been doing that in our project to try and make it clearer to users when we break things.

Reply:
No. I consider all three of those to "break" the API in some sense. So that section to me is just "anything that affects existing API in a way that users need to take action". I include deprecation warnings here already because that may break test suites, etc.

Reply:
But I agree, grouping in subsections can be useful if there are many items.

Reply:
+1 for keeping initial deprecations separate from removals / completed deprecations.

Reply:
Otherwise I think Lars' suggestions sound good, although I would expand "new features" to include (significant) performance improvements rather than just API additions. Maybe "enhancements" as a more general term (unless that is too general).